File: crates/ubaa-core/src/domain/libbook.rs
use serde::{Deserialize, Serialize};

use super::ActionEligibility;

#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookLibrary {
    pub id: String,
    pub name: String,
    pub free_num: i32,
    pub total_num: i32,
    pub storeys: Vec<LibBookStorey>,
}
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookStorey {
    pub id: String,
    pub name: String,
    pub free_num: i32,
    pub total_num: i32,
}
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookArea {
    pub id: String,
    pub name: String,
    pub area_name: String,
    pub premises_id: String,
    pub storey_id: String,
    pub free_num: i32,
    pub total_num: i32,
}
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookAreaDetail {
    pub id: String,
    pub name: String,
    pub available_dates: Vec<String>,
    pub time_slots: Vec<LibBookTimeSlot>,
}
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookTimeSlot {
    pub id: String,
    pub start: String,
    pub end: String,
    pub label: String,
}
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookSeat {
    pub id: String,
    pub name: String,
    pub no: String,
    pub status: Option<i32>,
    pub status_name: String,
    pub reserve_eligibility: ActionEligibility,
    pub reserve_target: Option<String>,
}
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookBooking {
    pub id: String,
    pub name_merge: String,
    pub area_name: String,
    pub seat_no: String,
    pub day: String,
    pub begin_time: String,
    pub end_time: String,
    pub status: Option<i32>,
    pub status_name: String,
    pub cancel_eligibility: ActionEligibility,
    pub cancel_target: Option<String>,
}
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookBookingsPage {
    pub bookings: Vec<LibBookBooking>,
    pub page: i32,
    pub limit: i32,
    pub total: i32,
}
/// 图书馆座位预约请求。
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookReserveRequest {
    pub area_id: String,
    pub seat_id: String,
    pub day: String,
    pub segment: String,
    pub start_time: String,
    pub end_time: String,
}
/// 图书馆座位预约的当前权威摘要。
///
/// 该值只包含稳定目标与脱敏展示字段，不包含业务令牌、Cookie 或原始上游响应。
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookReservePreflight {
    pub area_id: String,
    pub seat_id: String,
    pub seat_name: String,
    pub seat_no: String,
    pub day: String,
    pub segment: String,
    pub start_time: String,
    pub end_time: String,
}
/// 图书馆预约结果。
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookReserveResult {
    pub success: bool,
    pub message: String,
    pub booking: Option<LibBookBooking>,
}
/// 图书馆预约取消请求。
///
/// `page` 与 `limit` 固定 action 产生时所在分页，使 prepare/commit 都只在同一页 fresh 复核。
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookCancelRequest {
    pub booking_id: String,
    pub page: i32,
    pub limit: i32,
}
/// 图书馆预约取消的当前权威摘要。
///
/// 该值不包含 bearer、Cookie、原始响应或调用方不能核验的隐藏状态。
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookCancelPreflight {
    pub booking_id: String,
    pub booking_name: String,
    pub area_name: String,
    pub seat_no: String,
    pub day: String,
    pub begin_time: String,
    pub end_time: String,
}
/// 图书馆取消结果。
#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LibBookCancelResult {
    pub success: bool,
    pub message: String,
}

File: packages/ubaa_domain/lib/src/feature/presentation/libbook.dart
part of '../presentation.dart';

// 原始公开值与读取归属；写入资格和目标始终位于独立action。

final class LibbookLibraryPresentation extends FeaturePresentation {
  const LibbookLibraryPresentation({
    required this.id,
    required this.name,
    required this.freeNum,
    required this.totalNum,
    required this.queryDate,
    required this.storeys,
  });
  final String id;
  final String name;
  final int freeNum;
  final int totalNum;
  final String queryDate;
  final List<LibbookStoreyPresentation> storeys;
}

final class LibbookStoreyPresentation {
  const LibbookStoreyPresentation({
    required this.id,
    required this.name,
    required this.freeNum,
    required this.totalNum,
  });
  final String id;
  final String name;
  final int freeNum;
  final int totalNum;
}

final class LibbookAreaPresentation extends FeaturePresentation {
  const LibbookAreaPresentation({
    required this.id,
    required this.name,
    required this.areaName,
    required this.premisesId,
    required this.storeyId,
    required this.freeNum,
    required this.totalNum,
    required this.queryDate,
  });
  final String id;
  final String name;
  final String areaName;
  final String premisesId;
  final String storeyId;
  final int freeNum;
  final int totalNum;
  final String queryDate;
}

final class LibbookAreaDetailPresentation extends FeaturePresentation {
  const LibbookAreaDetailPresentation({
    required this.id,
    required this.name,
    required this.availableDates,
    required this.timeSlots,
  });
  final String id;
  final String name;
  final List<String> availableDates;

  /// 公开合同没有日期与时段关联，不得据此自动给某个日期配对。
  final List<LibbookTimeSlotPresentation> timeSlots;
}

final class LibbookTimeSlotPresentation {
  const LibbookTimeSlotPresentation({
    required this.id,
    required this.start,
    required this.end,
    required this.label,
  });
  final String id;
  final String start;
  final String end;
  final String label;
}

final class LibbookSeatPresentation extends FeaturePresentation {
  const LibbookSeatPresentation({
    required this.id,
    required this.name,
    required this.number,
    required this.status,
    required this.statusName,
    required this.areaId,
    required this.queryDate,
    required this.segment,
    required this.startTime,
    required this.endTime,
  });
  final String id;
  final String name;
  final String number;
  final int? status;
  final String statusName;
  final String areaId;
  final String queryDate;
  final String segment;
  final String startTime;
  final String endTime;
}

final class LibbookBookingPresentation extends FeaturePresentation {
  const LibbookBookingPresentation({
    required this.id,
    required this.name,
    required this.areaName,
    required this.seatNumber,
    required this.day,
    required this.beginTime,
    required this.endTime,
    required this.status,
    required this.statusName,
  });
  final String id;
  final String name;
  final String areaName;
  final String seatNumber;
  final String day;
  final String beginTime;
  final String endTime;
  final int? status;
  final String statusName;
}
